crates/polars-json/src/ndjson/deserialize.rs

```diff
@@ -14,27 +16,44 @@ use super::*;
 pub fn deserialize_iter<'a>(
     rows: impl Iterator<Item = &'a str>,
     data_type: ArrowDataType,
-    buf_size: usize,
-    count: usize,
 ) -> PolarsResult<ArrayRef> {
-    let mut buf = String::with_capacity(buf_size + count + 2);
+    let mut arr: Vec<Box<dyn Array>> = Vec::new();
+    let mut buf = String::with_capacity(std::u32::MAX as usize);
```

Comment:
This should be `std::cmp::min(buf_size + count + 2, std::u32::MAX as usize))`

As it is now, it always allocates 4GB of bytes!

Reply:
Ah good point!